nexus/src/external_api/http_entrypoints.rs

```diff
@@ -704,6 +710,40 @@ async fn project_disks_delete_disk(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+/// Create a snapshot of a disk.
+#[endpoint {
+    method = POST,
+    path = "/organizations/{organization_name}/projects/{project_name}/disks/{disk_name}/snapshot",
+    tags = ["disks"],
```

Comment:
Shouldn't this also be tagged w/ `snapshots`?

Reply:
I agree, and along those lines, what about 

```
POST /organizations/{organization_name}/projects/{project_name}/snapshots
```

with the name or ID of the disk in the body? Just feels weird to be separate

Reply:
> Shouldn't this also be tagged w/ `snapshots`?

Good catch, updated - this seems to fit the pattern elsewhere of "in a POST request, the tag is the thing being created".

> I agree, and along those lines, what about
> ```
> POST /organizations/{organization_name}/projects/{project_name}/snapshots
> ```
> with the name or ID of the disk in the body? Just feels weird to be separate

This part I'm less sure about. In comparison, if we look at the relationship of VPCs to subnets, the subnet object is created from the VPC, and listed as a path parameter:

https://github.com/oxidecomputer/omicron/blob/0eff1af73ef97f4370d2525d2773d1a58f75fe0f/nexus/src/external_api/http_entrypoints.rs#L1364-L1376

I figured this matched that pattern - and also matched the guidance from [RFD 4's guidance on snapshot APIs](https://rfd.shared.oxide.computer/rfd/0004#_snapshots) (although admittedly, things changed from that original spec by being nested under orgs + using names instead of IDs).

What's the decision-making criteria on whether or not to pass something as a path parameter or as a part of the body?
They seem functionally identical, but I wanna do what I can to make the API coherent and consistent.

Reply:
The GET endpoints for subnets also hang off the VPC, though, so it's consistent — all the subnet endpoints are in the same spot.

In this case, snapshots do not depend on the original disk once they're created, do they? The disk can go away and the snapshot will still exist. If that's true, my thought process on deciding what's a path param and what's in the body is simply to start from the path that makes sense to me (which I think is `/organizations/{organization_name}/projects/{project_name}/snapshots`). If that's the path, the disk ID/name has to be in the body because it can't go anywhere else.

If the snapshot _does_ depend on the disk forever, then I would lean the other way — all the endpoint paths should refer to the disk.

Reply:
> If the snapshot does depend on the disk forever, then I would lean the other way — all the snapshot paths should refer to the disk.

This is definitely a murky area, and it's (to some degree) up to us. The current API (added as a part of this PR) exposes a view of the `Snapshot` object which includes the source disk UUID forever.

Internally, however, the snapshot will point to a "volume" object, which represents the disk at that exact point in time, but the disk from which the snapshot was originally taken can continue to be modified (and will be represented by a different "Volume" object altogether, as it continues to mutate and potentially grow).

Reply:
If you're on the fence — we talked about this at the console team sync and we all agree pretty strongly that snapshot create should be at the same path as get (`/orgs/:org/projects/:project/snapshots`), which means disk ID has to be a create-time body parameter.

Reply:
Okay, related to this in the ongoing name/ID battle - even if I move the "disk" into the body params, does it remain a "Name" or an "ID"? (Or an enum to be either?)

Reply:
(For the moment, I'm going with "Name", since that seems kinda like the default choice)